src/cascadia/TerminalSettingsModel/Command.h

```diff
@@ -94,11 +97,14 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
                                                                                            bool directories,
                                                                                            hstring iconPath);
 
+        WINRT_PROPERTY(Model::ActionAndArgs, ActionAndArgs, Model::ActionAndArgs{});
         WINRT_PROPERTY(ExpandCommandType, IterateOn, ExpandCommandType::None);
-        WINRT_PROPERTY(Model::ActionAndArgs, ActionAndArgs);
         WINRT_PROPERTY(OriginTag, Origin);
         WINRT_PROPERTY(winrt::hstring, Description, L"");
 
+    public:
+        til::typed_event<Model::Command, winrt::hstring> IDChanged;
```

Comment:
this could be expensive (memory-wise) since every command will now have a fully populated eventh handler  object sitting on it. we don't do it that way for color schemes -- we actually make `CascadiaSettings` handle renaming them